community/kernel/src/test/java/org/neo4j/kernel/impl/query/Neo4jTransactionalContextTest.java

```diff
@@ -154,4 +154,78 @@ private void setUpMocks()
         when( dependencyResolver.resolveDependency( ThreadToStatementContextBridge.class ) ).thenReturn( statementContextBridge );
         when( dependencyResolver.resolveDependency( Guard.class ) ).thenReturn( guard );
     }
+
+    @SuppressWarnings( "ConstantConditions" )
+    @Test
+    public void rollsBackNewlyCreatedTransactionIfTerminationDetectedOnCloseDuringPeriodicCommit()
+    {
+        // Given
+        GraphDatabaseQueryService graph = mock( GraphDatabaseQueryService.class );
```

Comment:
That would be bad practice. If this fails, we should stop and think why.  I understand that this is not perfect yet it is the only good test we felt we could write here. We didn't find a better way to test the behavior we need to rely on in the built in procedures without resorting to heavy use of mocking or probabilistic (yuck) testing. How would you test this?